### PR TITLE
Create 5 Vehicle::CableLiftUpdate... member functions from cable_lift_update_... (#9473)

### PR DESCRIPTION
**Pull request by mustaphaelghoul**
Updated:
      cable_lift_update_moving_to_end_of_station to Vehicle::CableLiftUpdateMovingToEndOfStation
      cable_lift_update_waiting_to_depart to Vehicle::CableLiftUpdateWaitingToDepart
      cable_lift_update_departing to Vehicle::CableLiftUpdateDeparting
      cable_lift_update_travelling to Vehicle::CableLiftUpdateTravelling
      cable_lift_update_arriving to to Vehicle::CableLiftUpdateArrriving

### CODE REVIEW

File: src/openrct2/ride/Vehicle.h [hunk 1]
```diff
@@ -350,6 +350,11 @@ struct Vehicle : SpriteBase
     bool CurrentTowerElementIsTop();
     bool UpdateTrackMotionForwards(rct_ride_entry_vehicle* vehicleEntry, Ride* curRide, rct_ride_entry* rideEntry);
     bool UpdateTrackMotionBackwards(rct_ride_entry_vehicle* vehicleEntry, Ride* curRide, rct_ride_entry* rideEntry);
+    void CableLiftUpdateMovingToEndOfStation();
+    void CableLiftUpdateWaitingToDepart();
+    void CableLiftUpdateDeparting();
+    void CableLiftUpdateTravelling();
+    void CableLiftUpdateArriving();
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.h:357]: Move these to just below `CableLiftUpdate()` so alike functions are grouped together.

**Comment by mustaphaelghoul** [comment]
> Seems ok, just a minor comment.
> 
> I'd have made each function move in a separate commit as well, just for the sake of following a "1 change per commit" rule. This PR would then have 5 commits, but the same diff :)

Whoops, saw that too late :p

File: src/openrct2/ride/Vehicle.h (hunk 1 above)
Comment (mustaphaelghoul) [reply src/openrct2/ride/Vehicle.h]: done 👍

File: src/openrct2/ride/CableLift.cpp [hunk 2]
```diff
@@ -119,104 +113,104 @@ void Vehicle::CableLiftUpdate()
...
 
     // Next check to see if the second part of the cable lift
     // is at the front of the passenger vehicle to simulate the
     // cable being attached underneath the train.
-    Vehicle* passengerVehicle = GET_VEHICLE(vehicle->cable_lift_target);
-    Vehicle* cableLiftSecondPart = GET_VEHICLE(vehicle->prev_vehicle_on_ride);
+    Vehicle* passengerVehicle = GET_VEHICLE(cable_lift_target);
+    Vehicle* cableLiftSecondPart = GET_VEHICLE(prev_vehicle_on_ride);
 
     int16_t dist_x = abs(passengerVehicle->x - cableLiftSecondPart->x);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/CableLift.cpp:158]: Could you turn these two variables to camelCase

File: src/openrct2/ride/CableLift.cpp [hunk 3]
```diff
@@ -119,104 +113,104 @@ void Vehicle::CableLiftUpdate()
...
 
-    vehicle->velocity = std::min(passengerVehicle->velocity, 439800);
-    vehicle->acceleration = 0;
+    velocity = std::min(passengerVehicle->velocity, 439800);
+    acceleration = 0;
     if (passengerVehicle->update_flags & VEHICLE_UPDATE_FLAG_BROKEN_TRAIN)
         return;
 
-    if (!(cable_lift_update_track_motion(vehicle) & (1 << 1)))
+    if (!(cable_lift_update_track_motion(this) & (1 << 1)))
```
Comment (duncanspumpkin) [inline src/openrct2/ride/CableLift.cpp:197]: `(1<<1)` in this instance is `VEHICLE_UPDATE_MOTION_TRACK_FLAG_1` could you make this change as well.

File: src/openrct2/ride/CableLift.cpp [hunk 4]
```diff
@@ -119,104 +113,104 @@ void Vehicle::CableLiftUpdate()
...
+    if (velocity < -439800)
     {
-        vehicle->velocity -= vehicle->velocity / 16;
-        vehicle->acceleration = 0;
+        velocity -= velocity / 16;
+        acceleration = 0;
     }
 
-    if (!(cable_lift_update_track_motion(vehicle) & (1 << 0)))
+    if (!(cable_lift_update_track_motion(this) & (1 << 0)))
```
Comment (duncanspumpkin) [inline src/openrct2/ride/CableLift.cpp:127]: `VEHICLE_UPDATE_MOTION_TRACK_FLAG_VEHICLE_AT_STATION` for (1<<0)

**Comment by janisozaur** [comment]
I thought this was going to get squashed. The commit messages here are not of the expected standard.

**Pull request merged**